packages/PostRector/Rector/NameImportingPostRector.php

```diff
@@ -128,4 +137,13 @@ private function shouldImportName(Name $name, File $file, array $currentUses): b
 
         return $this->reflectionProvider->hasFunction(new Name($name->getLast()), null);
     }
+
+    private function shouldApply(File $file): bool
+    {
+        if ($file->hasContentChanged()) {
+            return true;
+        }
+
+        return ! $this->parameterProvider->provideBoolParameter(Option::APPLY_AUTO_IMPORT_NAMES_ON_CHANGED_FILES_ONLY);
```

Comment:
I think comparing content can be only when there is `APPLY_AUTO_IMPORT_NAMES_ON_CHANGED_FILES_ONLY` enabled:

```suggestion
       if (! (bool) $this->parameterProvider->provideBoolParameter(Option::APPLY_AUTO_IMPORT_NAMES_ON_CHANGED_FILES_ONLY)) {
            return true;
       }

       return $file->hasContentChanged();
```